tests/app/main/views/test_services.py

```diff
@@ -431,29 +431,111 @@ def test_service_update_documents(self, data_api_client):
     def test_service_update_documents_with_validation_errors(
             self, data_api_client):
         data_api_client.get_service.return_value = {'services': {
-            'id': 1,
+            'id': 7654,
             'supplierId': 2,
             'frameworkSlug': 'g-cloud-7',
             'lot': 'SCS',
-            'serviceDefinitionDocumentURL': "http://assets/documents/1/2-service-definition.pdf",  # noqa
-            'pricingDocumentURL': "http://assets/documents/1/2-pricing.pdf",
+            'serviceDefinitionDocumentURL': "http://assets/documents/7654/2-service-definition.pdf",  # noqa
+            'pricingDocumentURL': "http://assets/documents/7654/2-pricing.pdf",
             'sfiaRateDocumentURL': None
         }}
         response = self.client.post(
-            '/admin/services/1/edit/documents',
+            '/admin/services/7654/edit/documents',
             data={
                 'serviceDefinitionDocumentURL': (StringIO(), ''),
                 'pricingDocumentURL': (StringIO(b"doc"), 'test.pdf'),
                 'sfiaRateDocumentURL': (StringIO(b"doc"), 'test.txt'),
                 'termsAndConditionsDocumentURL': (StringIO(), 'test.pdf'),
             }
         )
+        document = html.fromstring(response.get_data(as_text=True))
 
-        data_api_client.get_service.assert_called_with('1')
+        data_api_client.get_service.assert_called_with('7654')
         assert data_api_client.update_service.called is False
+        assert response.status_code == 400
+
+        assert document.xpath(
+            "//*[contains(@class,'validation-message')][contains(normalize-space(string()), $t)]",
+            t="Your document is not in an open format",
+        )
+        assert document.xpath(
+            "//a[normalize-space(string())=$t]/@href",
+            t="Return without saving",
+        ) == ["/admin/services/7654"]
+
+    # TODO test_service_update_non_assurance_questions_when_API_returns_error?
```

Comment:
I think it's OK without this extra test.  But up to you - don't leave the `TODO` here though.